lib/src/rewrite.rs

```diff
@@ -150,21 +150,21 @@ pub fn rebase_commit_with_options(
     // Ensure we don't abandon commits with multiple parents (merge commits), even
     // if they're empty.
     if let [parent] = new_parents {
-        match options.empty {
-            EmptyBehaviour::AbandonNewlyEmpty | EmptyBehaviour::AbandonAllEmpty => {
-                if *parent.tree_id() == new_tree_id
-                    && (options.empty == EmptyBehaviour::AbandonAllEmpty
-                        || old_base_tree_id != Some(old_commit.tree_id().clone()))
-                {
-                    mut_repo.record_abandoned_commit(old_commit.id().clone());
-                    // Record old_commit as being succeeded by the parent for the purposes of
-                    // the rebase.
-                    // This ensures that when we stack commits, the second commit knows to
-                    // rebase on top of the parent commit, rather than the abandoned commit.
-                    return Ok(parent.clone());
-                }
+        let to_abandon = match options.empty {
```

Comment:
nit: The name `to_abandon` made me think it was a container of the commits to abandon. How about `should_abandon`?